interpreter/interpretable.go

```diff
@@ -439,14 +439,9 @@ func (un *evalUnary) ID() int64 {
 func (un *evalUnary) Eval(ctx Activation) ref.Val {
 	argVal := un.arg.Eval(ctx)
 	// Early return if the argument to the function is unknown or error.
-	if un.allowError {
-		if types.IsUnknown(argVal) {
-			return argVal
-		}
-	} else {
-		if types.IsUnknownOrError(argVal) {
-			return argVal
-		}
+	strict := !un.nonStrict
```

Comment:
These assignments are for ease of reading; the SSA pass should resolve them to the same code as the reduced form.